### DISCUSSION TITLE
Examples of python websocket client

### DISCUSSION THREAD

**Discussion by andrewwongagility** [foxglove/community#950] [Q&A] [answered]
Does foxglove provide any examples of a python websocket client that can connect to a Foxglove websocket? If not, is there any code I can review to help me get started for writing a python client? Thank you!

From my understanding the python client code would need to send the json messages shown [here](https://github.com/nbbrooks/ws-protocol/blob/288af41197f8302ee19b6936a72162ede4c990a5/docs/spec.md#sent-by-client)

**Comment by jtbandes** [comment, marked as answer]
Pretty much everything is in this repo: https://github.com/foxglove/ws-protocol
We don't have a Python client there today, but you might find the TypeScript client to be a reasonable starting point if you want to implement one.

**Comment by defunctzombie** [comment]
@andrewwongagility If you do make a client you feel comfortable sharing as an example or becoming the client library we'd be happy to review a PR to the ws-protocol repo so the community can have a shared library we all improve.

**Reply by andrewwongagility** [reply]
@defunctzombie That sounds good! I'll try to open a PR to add the Python client example when I'm finished with it
